### PR TITLE
JS: make sanitization a "common" technique rather than "important"

### PR DESCRIPTION
**Pull request by esbena**
It was pointed out that sanitization is bordering on being an anti-pattern, and that we should make it clearer in our documentation that better alternatives exist.
I have therefore changed the "sanitization is important" formulations to "sanitization is common", and added an example of an alternative where appropriate.

### CODE REVIEW

**Review by asgerf — APPROVED** [review]
I like it! 👍

**Comment by erik-krogh** [comment]
You could put the `An even safer alternative` texts first in the `<recommendation>`.  
(E.g. `IncompleteSanitization.qhelp`: the first proposal given in `<recommendation>` is to keep using sanitization).  

That would require a bit more rewriting, but I think it could be good to present the safest option first.

**Comment by esbena** [comment]
> but I think it could be good to present the safest option first.

I think the order is fine. We are generally recommending local fixes for the alerts we produce, not architectural rewrites.

Ping @mchammer01 for a minor doc review

File: javascript/ql/src/Security/CWE-116/IncompleteHtmlAttributeSanitization.qhelp
Comment (sj) [inline javascript/ql/src/Security/CWE-116/IncompleteHtmlAttributeSanitization.qhelp:44]: I'll leave it up to the documentation team to decide whether it's preferable to say "isn't" or "is not".

Comment (felicitymay) [inline javascript/ql/src/Security/CWE-116/IncompleteHtmlAttributeSanitization.qhelp:44]: I'd suggest that it's clearer as:

```suggestion
			so that sanitization is not needed, for instance by using HTML
```

File: javascript/ql/src/Security/CWE-116/IncompleteSanitization.qhelp
Comment (felicitymay) [inline javascript/ql/src/Security/CWE-116/IncompleteSanitization.qhelp:38]:
```suggestion
An even safer alternative is to design the application so that sanitization is not
needed, for instance by using prepared statements for SQL queries.
```

File: javascript/ql/src/Security/CWE-116/UnsafeHtmlExpansion.qhelp
Comment (felicitymay) [inline javascript/ql/src/Security/CWE-116/UnsafeHtmlExpansion.qhelp:35]:
```suggestion
			so that sanitization is not needed, for instance by using HTML
```

File: javascript/ql/src/Security/CWE-843/TypeConfusionThroughParameterTampering.qhelp
Comment (felicitymay) [inline javascript/ql/src/Security/CWE-843/TypeConfusionThroughParameterTampering.qhelp:42]:
```suggestion
			An even safer alternative is to design the application so that sanitization is not
			needed, for instance by using prepared statements for SQL queries.
```

File: javascript/ql/src/Security/CWE-116/IncompleteHtmlAttributeSanitization.qhelp
Comment (felicitymay) [reply javascript/ql/src/Security/CWE-116/IncompleteHtmlAttributeSanitization.qhelp:44]: While the original text adds emphasis to the statement for fluent speakers, it may confuse people who are less proficient.

**Review by felicitymay — DISMISSED** [review]
Thanks for making the help more precise and helpful to users. I've made one suggestion several times, but otherwise the changes LGTM once the tests pass.

**Comment by esbena** [comment]
Thank you all. I have applied @felicitymay's suggestions, and just need a final approval.

**Review by erik-krogh — APPROVED** [review]

**Comment by mchammer01** [comment]
@esbena - thanks for the ping and sorry for not responding (I was away yesterday.)
Thanks Felicity for looking at this.

**Pull request merged**
